app-creator/client/src/utils/test/helpers_test.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ *  @fileoverview This file tests the helper functions defined in utils.
+ */
+import { expect } from '@open-wc/testing';
+import { AppCreatorStore } from '../../redux/reducer';
+import { PaletteNames } from '../../redux/types/enums';
+import {
+  deepCloneTemplate,
+  camelCaseToTitleCase,
+  getWidgetType,
+  getIdPrefixLastIndex,
+  generateRandomId,
+} from '../helpers';
+
+suite('util helpers', () => {
+  suite('deep cloning templates', () => {
+    test('clones all values', () => {
+      const template = createTemplateStub();
+      const templateCopy = deepCloneTemplate(template);
+
+      expect(templateCopy).deep.equal(template);
+    });
+
+    test('creates new references for object types (including arrays)', () => {
+      const template = createTemplateStub();
+      const templateCopy = deepCloneTemplate(template);
+
+      expect(templateCopy.widgets['panel-0'].children).not.equal(
+        template.widgets['panel-0'].children
+      );
```

Comment:
Re:comment above, we'll want a test to check that the elements of the children aren't referring to the same objects.

Reply:
In the Redux store, all array elements contain primitive data types, so checking if the array reference itself is different should be sufficient.